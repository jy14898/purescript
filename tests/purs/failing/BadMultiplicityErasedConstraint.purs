-- @shouldFailWith NoInstanceFound
module Main where

import Prelude
import Effect.Console (log)

data X = X

class Y a where
  member :: a -> Unit

instance yX :: Y X where
  member X = unit

test1 :: forall a. Y a ? a -> Unit
test1 a = member a

main = log "Done"
