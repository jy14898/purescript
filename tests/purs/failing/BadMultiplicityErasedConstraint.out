Error found:
in module [33mMain[0m
at tests/purs/failing/BadMultiplicityErasedConstraint.purs:16:11 - 16:19 (line 16, column 11 - line 16, column 19)

  No type class instance was found for
  [33m           [0m
  [33m  Main.Y a1[0m
  [33m           [0m

while applying a function [33mmember[0m
  of type [33mY t0 => t0 -> Unit[0m
  to argument [33ma[0m
while checking that expression [33mmember a[0m
  has type [33mUnit[0m
in value declaration [33mtest1[0m

where [33ma1[0m is a rigid type variable
        bound at (line 0, column 0 - line 0, column 0)
      [33mt0[0m is an unknown type

See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
or to contribute content related to this error.

